src/probes/journal.c

```diff
@@ -224,10 +224,14 @@ static bool add_filters(sd_journal *journal)
 
         MATCH(data);
         free(data);
+        AND;
         // The three highest log levels, all indicating errors
         MATCH("PRIORITY=1");
         MATCH("PRIORITY=2");
         MATCH("PRIORITY=3");
+        OR;
```

Comment:
Perhaps name-space the macros (JOURNAL_MATCH), (JOURNAL_AND) etc... 

Have a comment summarizing boolean logic (boot_id & (1 || 2 || 3 || exited)

Unit tests?